logic/LogicManager.java

```diff
@@ -103,6 +106,14 @@ private FunctionalIterator<Rule> rulesWithNegations() {
         return rules().filter(rule -> !rule.when().negations().isEmpty());
     }
 
+    public FunctionalIterator<Unifier> unifiers(Concludable concludable, Rule rule) {
+        return iterate(applicableRules(concludable).getOrDefault(rule, new HashSet<>()));
+    }
+
+    public Map<Rule, Set<Unifier>> applicableRules(Concludable concludable) {
+        return logicCache.applicableRules().get(concludable, c -> c.applicableRules(conceptMgr, this));
```

Comment:
choose either just `applicableRules` returning the `Map`, or two APIs, one which returns `Rules` and the other which returns `Unifiers`